src/lib/staged_ledger/staged_ledger.ml

```diff
@@ -1983,11 +1983,89 @@ module T = struct
         : Ledger.unattached_mask ) ;
     r
 
+  module Application_state = struct
+    type txn =
+      ( Signed_command.With_valid_signature.t
+      , Zkapp_command.Valid.t )
+      User_command.t_
+
+    type t =
+      { valid_seq : txn Sequence.t
+      ; invalid : (txn * Error.t) list
+      ; skipped_by_fee_payer : txn list Account_id.Map.t
```

Comment:
NIT: We never read txns stored in this map. It doesn't hurt much to track this information, but it might make more sense to have a set of accounts for which we have skipped txns.

Reply:
It seems like it is not needed?